connectors/sources/google_drive.py

```diff
@@ -515,8 +583,7 @@ def get_default_configuration(cls):
             },
         }
 
-    @cached_property
-    def google_drive_client(self):
+    def google_drive_client(self, impersonate_email=None):
```

Comment:
Maybe we can add a small comment explaining why this is not cached and why we need to reinstantiate?